XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -2012,5 +2014,20 @@ private void ToggleExpandedState(Func<PDSection, bool> expand)
                 panel2.ResumeLayout();
             }
         }
+
+        private void SetupDeprecationBanner()
+        {
+            Banner.Visible = false;

```

Comment:
When the cache is updated, this method maybe called more than once, so always hiding it before checking whether to re-show it will cause flickering. How about this:
```
if (Helpers.PostStockholm(xenObject.Connection) || (int)API_Version.LATEST < (int)API_Version.API_2_16)
{
    Banner.Visible = false;
}
else
{
    ...
    Banner.Visible = true;
}
```

Reply:
Fixed.